src/Rendering/Renderer.cpp

```diff
@@ -24,26 +24,37 @@ Renderer::Renderer(const flecs::world& ecs)
 			.kind(flecs::PreStore)
 			.each(PreDraw);
 
+
+	//QUESTION: had to change it to OnUpdate cause map was drawn over rectangles
```

Comment:
Which rectangles? The healthbars? If yes, it might be more `UpdateHealthBar` that needs to be moved to `flecs::OnStore`. You could still split the logic of that system in two systems (one to compute the positions of the healthbars in `PostUpdate` and store the results in a singleton, and have a second system that only takes care of drawing healthbars at the positions stored in that singleton)